zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -1014,3 +1015,45 @@ def test_invalid_compression_configuration(self):
                          'configuration')
             self.test_config[
                 'target_deploy_status'] = stored_target_deploy_status
+
+
+class CephDepartureTest(test_utils.OpenStackBaseTest):
+    """Test charm handling of a departing unit."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running ceph departure tests."""
+        super(CephDepartureTest, cls).setUpClass()
+
+    def get_mon_count(self):
+        """Compute the number of active ceph monitors."""
+        unit = zaza_model.get_units('ceph-mon')[0].entity_id
+        result = zaza_model.run_on_unit(unit, 'ceph mon stat')
+        rx = re.compile('(\\d)* mons')
+        match = rx.search(result['Stdout'])
+        return int(match.group(1))
+
+    def get_mon_unit(self):
+        """Pick a non-leader ceph-mon unit to run a command on."""
+        status = zaza_model.get_status().applications['ceph-mon']
+        for unit in status['units']:
+            if not status['units'][unit].get('leader'):
+                return unit
+
+    def test_departure(self):
+        """Test descaling of a ceph-mon unit."""
+        mon_count = self.get_mon_count()
+        logging.info('Monitor count is {}'.format(mon_count))
+        # We add a new unit so that the ceph-mon count remains
+        # above the 'monitor-count' specified in the bundle.
+        zaza_model.add_unit('ceph-mon')
+        logging.info('Waiting for new ceph-mon unit to settle')
+        zaza_model.block_until_unit_count('ceph-mon', mon_count + 1)
+        zaza_model.block_until_all_units_idle()
+
+        # Now remove a ceph-mon unit and test for cluster consistency.
+        zaza_model.destroy_unit('ceph-mon', self.get_mon_unit())
```

Comment:
Given that the Juju leader for a deployed ceph-mon doesn't really relate to the Ceph cluster, I'd be tempted to suggest grabbing the first mon, rather than filtering by leadership